react/features/recording/components/LiveStream/AbstractStartLiveStreamDialog.js

```diff
@@ -0,0 +1,337 @@
+                }
+            </Dialog>
+        );
+    }
+
+    _onCancel: () => boolean;
+
+    /**
+     * Invokes the passed in {@link onCancel} callback and closes
+     * {@code StartLiveStreamDialog}.
+     *
+     * @private
+     * @returns {boolean} True is returned to close the modal.
+     */
+    _onCancel() {
+        sendAnalytics(createRecordingDialogEvent('start', 'cancel.button'));
+
+        return true;
+    }
+
+    /**
+     * Asks the user to sign in, if not already signed in, and then requests a
+     * list of the user's YouTube broadcasts.
+     *
+     * NOTE: To be implemented by platforms.
+     *
+     * @private
+     * @returns {Promise}
+     */
+    _onGetYouTubeBroadcasts: () => Promise<*>;
+
+    /**
+     * Method to be implemented by platform specific components.
+     */
+    _onInitializeGoogleApi: () => Object;
+
+    _onStreamKeyChange: string => void;
+
+    /**
+     * Callback invoked to update the {@code StartLiveStreamDialog} component's
+     * display of the entered YouTube stream key.
+     *
+     * @param {string} streamKey - The stream key entered in the field.
+     * changed text.
+     * @private
+     * @returns {void}
+     */
+    _onStreamKeyChange(streamKey) {
+        this._setStateIfMounted({
+            streamKey,
+            selectedBoundStreamID: undefined
+        });
+
+        this.props.dispatch(setLiveStreamKey(streamKey));
+    }
+
+    _onSubmit: () => boolean;
+
+    /**
+     * Invokes the passed in {@link onSubmit} callback with the entered stream
+     * key, and then closes {@code StartLiveStreamDialog}.
+     *
+     * @private
+     * @returns {boolean} False if no stream key is entered to preventing
+     * closing, true to close the modal.
+     */
+    _onSubmit() {
+        const { broadcasts, selectedBoundStreamID } = this.state;
+        const key = this.state.streamKey || this.props._streamKey;
+
+        if (!key) {
+            return false;
+        }
+
+        let selectedBroadcastID = null;
+
+        if (selectedBoundStreamID) {
+            const selectedBroadcast = broadcasts && broadcasts.find(
+                broadcast => broadcast.boundStreamID === selectedBoundStreamID);
+
+            selectedBroadcastID = selectedBroadcast && selectedBroadcast.id;
+        }
+
+        sendAnalytics(createRecordingDialogEvent('start', 'confirm.button'));
+
+        this.props._conference.startRecording({
+            broadcastId: selectedBroadcastID,
+            mode: JitsiRecordingConstants.mode.STREAM,
+            streamId: key
+        });
+
+        return true;
+    }
+
+    /**
+     * Updates the internal state if the component is still mounted. This is a
+     * workaround for all the state setting that occurs after ajax.
+     *
+     * @param {Object} newState - The new state to merge into the existing
+     * state.
+     * @private
+     * @returns {void}
+     */
+    _setStateIfMounted(newState) {
```

Comment:
This is indeed older web code, but I think the same issue can happen on mobile too, as the content of the component can be updated asynchronously by the native Google api (e.g. after sign in, or after fetching streams) and the component may be unmounted by then.

Reply:
I don't think a field (regardless of web or mobile) can request state update after unmounted, that is only for Google API.

If you're wondering however why it's there as there is no Google Sign-in on mobile, you are absolutely right, but there was a comment somewhere in the class explaining that although there is no Google API on mobile yet, it will surely come and this abstraction is already prepared for that for when it happens.